src/shared/context/auth-context.js

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, useContext, useEffect, useReducer } from 'react';
+import { authReducer } from '../reducer';
+import { authState } from './auth-state';
+import { INITIALIZE_AUTH } from '../actions/types';
+
+const AuthContext = createContext({});
+
+const AuthProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(authReducer, authState);
```

Comment:
You can use authstate and authdispatch as state and dispatch will be confusing if you are planning to use multiple reducers

Reply:
Renaming it while using in other components!